src/Pages/ControlPanel.jsx

```diff
@@ -25,24 +25,46 @@ const ControlPanel = () => {
 	const isSmall = useMediaQuery(theme.breakpoints.down("md"));
 
 	const [ctrlVals, dispatchCtrlVals] = useReducer(ctrlValsReducer, { remote: null, local: {} });
-	const fetchRemoteValues = (discardLocal) => {
-		dispatchCtrlVals({ type: "discard_remote" });
-		Promise.all(
-			systemControlsCollections.map(collName => getDocs(collection(db, collName)))
-		).then(querySnapshotArray => {
-			// get an array of arrays of [path, data] pairs, then flatten it to a array of [path, data] pairs
-			const pairs = querySnapshotArray.map(querySnapshot =>
-				// get an array of [path, data] pairs
-				querySnapshot.docs.map(doc => [doc.ref.path, doc.data()])
-			).flat();
-			if (discardLocal) {
-				// do this first to avoid the unnecessary checks in replace_remote
-				dispatchCtrlVals({ type: "discard_local" });
-			}
-			dispatchCtrlVals({ type: "replace_remote", newRemote: Object.fromEntries(pairs) });
-		});
-	};
-	useEffect(() => fetchRemoteValues(), []);
+
+  const fetchRemoteValues = (discardLocal) => {
+    useEffect(() => {
+      dispatchCtrlVals({ type: "discard_remote" });
+
+      
+      //this is the listener for the realtime updates
+      systemControlsCollections.map((collName) => {
+        const collectionRef = query(collection(db, collName));
+        console.log(collName)
+          const unsubscribe = onSnapshot(collectionRef, (querySnapshot) => {
+            const pairs = querySnapshot.docs.map(doc => [doc.ref.path, doc.data()]);
+            if (discardLocal) {
+              // do this first to avoid the unnecessary checks in replace_remote
+              dispatchCtrlVals({ type: "discard_local" });
+            }
+            dispatchCtrlVals({ type: "replace_remote", newRemote: Object.fromEntries(pairs) });
+          });
+          return () => unsubscribe();
+      });
+
+      //this gets the intial values
```

Comment:
Again I had the same first instinct, and i'm not sure why the onSnapshot block isn't sufficient. 

But basically if you run the code without the 2nd block, everything will initialize to empty values. But if you go into the db and change a value there, THEN it will trigger the listener and populate the values. Obviously we want it to initialize correctly. That's why I added the 2nd block.